tests/func/remote/test_gdrive.py

```diff
@@ -0,0 +1,56 @@
+import os
+import posixpath
+
+import configobj
+
+from dvc.compat import fspath
+from dvc.main import main
+from dvc.remote import GDriveRemote
+from dvc.repo import Repo
+
+
+def test_relative_user_credentials_file_config_setting(tmp_dir, dvc):
+    # CI sets it to test GDrive, here we want to test the work with file system
+    # based, regular credentials
+    if os.getenv(GDriveRemote.GDRIVE_CREDENTIALS_DATA):
+        del os.environ[GDriveRemote.GDRIVE_CREDENTIALS_DATA]
+
+    credentials = os.path.join("secrets", "credentials.json")
+
+    # GDriveRemote.credentials_location helper checks for file existence,
+    # create the file
+    tmp_dir.gen(credentials, "{'token': 'test'}")
+
+    remote_name = "gdrive"
+    assert (
+        main(["remote", "add", "-d", remote_name, "gdrive://root/test"]) == 0
+    )
+    assert (
+        main(
+            [
+                "remote",
+                "modify",
+                remote_name,
+                "gdrive_user_credentials_file",
+                credentials,
+            ]
+        )
+        == 0
+    )
+
+    # We need to load repo again to test updates to the config
+    str_path = fspath(tmp_dir)
+    repo = Repo(str_path)
+
```

Comment:
Probably won't be needed if you use API as requested above.

Reply:
I'll check, but most likely we'll need it anyway ... we change remote config and need initialize Repo again

we can for sure just initialize remote directly ... again, will be most like a unit test then